warehouse/upload.go

```diff
@@ -1776,29 +1775,28 @@ func (job *UploadJobT) createLoadFiles(generateAll bool) (startLoadFileID, endLo
 				DestinationRevisionID:        job.warehouse.Destination.RevisionID,
 				StagingDestinationRevisionID: stagingFile.DestinationRevisionID,
 			}
-
 			if revisionConfig, ok := destinationRevisionIDMap[stagingFile.DestinationRevisionID]; ok {
 				payload.StagingDestinationConfig = revisionConfig.Config
 			}
-
 			if misc.ContainsString(warehouseutils.TimeWindowDestinations, job.warehouse.Type) {
 				payload.LoadFilePrefix = warehouseutils.GetLoadFilePrefix(stagingFile.TimeWindow, job.warehouse)
 			}
 
-			// set merged schema as upload schema if the load file type is parquet
-			if job.upload.LoadFileType == warehouseutils.LOAD_FILE_TYPE_PARQUET {
-				payload.UploadSchema = job.upload.MergedSchema
-			}
-
 			payloadJSON, err := json.Marshal(payload)
 			if err != nil {
 				panic(err)
 			}
 			messages = append(messages, payloadJSON)
 		}
+		var schema *warehouseutils.SchemaT
+		if job.upload.LoadFileType == warehouseutils.LOAD_FILE_TYPE_PARQUET {
+			schema = &job.upload.MergedSchema
+		} else {
+			schema = &job.upload.UploadSchema
+		}
```

Comment:
I would rephrase it like this to better represent the intent and save 1 extra indentation,
```
schema := &job.upload.UploadSchema

if job.upload.LoadFileType == warehouseutils.LOAD_FILE_TYPE_PARQUET {
    schema = &job.upload.MergedSchema
}
```

Wdyt ? ( Just an opinion and not a hard stance )